patches/kaggle_bashrc

```diff
@@ -0,0 +1,31 @@
+
+# Kaggle-specific .bashrc script to be appended to /etc/bash.bashrc to apply it
+# when any user session starts.
+# This script sets up pip to enable a user to install and use python modules via
+# pip intall. $KAGGLE_WORKING_DIR should be available for it to work.
+
+if [[ ! -z "${KAGGLE_WORKING_DIR}" ]]; then
+    PIP_INSTALL_PREFIX_DIR="${KAGGLE_WORKING_DIR}/pip"
+    PIP_CONFIG_FILE_PATH="${KAGGLE_WORKING_DIR}/config/pip/pip.conf"
+
+    # Create a directory for pip to install modules.
+    mkdir -p ${PIP_INSTALL_PREFIX_DIR}
+
+    # Create pip config file to use the prefix directory created above for
+    # installation. Also, ignore-installed is set to true to prevent pip
+    # from trying to remove existing modules from the read-only filesystem.
+    mkdir -p `dirname ${PIP_CONFIG_FILE_PATH}`
+    echo "[install]\nprefix=${PIP_INSTALL_PREFIX_DIR}\nignore-installed=true" > ${PIP_CONFIG_FILE_PATH}
+    # Instruct pip to use this config file.
+    export PIP_CONFIG_FILE=${PIP_CONFIG_FILE_PATH}
+
+    # Set up PYTHONPATH correctly to include the user-installed library.
+    # Note that the pip prefix directory overrides the system default to enable
+    # a user to use his/her installed one.
+    # TODO(dsjang): Currently "lib/python3.6/site-packages" is hard-coded
+    # throughout Dockerfile. Parameterize it to avoid a version mismatch. 
+    export PYTHONPATH=${PIP_INSTALL_PREFIX_DIR}/lib/python3.6/site-packages:${PYTHONPATH}
```

Comment:
+1 on the parameterization in future. Could probably read in python version and interpolate I'm assuming though unsure if we'll support multiple versions or chicken / egg scenario with python command being dependent on PYTHONPATH hah.

Reply:
Another thing to possibly investigate in the future (if you haven't already) is leaving the top level implementation alone for reserved read-only control and instead setting `PYTHONUSERBASE`, then documenting to users that they should use:
```
!pip install --user ...
```
Unsure on any gotchas that might be relevant there. Also using `virtualenv` to put users into their own sandbox within the working directory is another option that might be worth investigating at some point.

https://stackoverflow.com/a/29103053/769871

Reply:
Acked. Thanks for the very useful feedback.

Your suggested method was actually what I wanted to use, but unfortunately we're using PYTHONUSERBASE for monkeypatching something while python is first loaded. I stashed your comment in case we want to clean things up better in the future.

Thanks!